lib/miq_automation_engine/engine/miq_ae_service.rb

```diff
@@ -237,6 +237,42 @@ def execute(m, *args)
       raise MiqAeException::MethodNotFound, err.message
     end
 
+    def notification_subject(values_hash)
+      if values_hash[:subject].present?
+        subject = values_hash[:subject]
+        subject = subject.instance_variable_get('@object') if values_hash[:subject].kind_of?(MiqAeMethodService::MiqAeServiceModelBase)
+        raise ArgumentError, "Subject must be a valid Active Record object" if !subject.kind_of?(ActiveRecord::Base)
+      else
+        subject = @workspace.ae_user
+      end
+      subject
+    end
+
+    def notification_type(values_hash)
+      if values_hash[:type].present? && !values_hash[:type].nil?
+        type = values_hash[:type].to_sym
+      else
+        level = values_hash[:level] || "info"
+        audience = values_hash[:audience] || "user"
+        _log.info("Generic notification type level: #{level} audience: #{audience}")
+        type = "automate_#{audience}_#{level}".downcase.to_sym
+      end
+      _log.info("Validating Notification type: #{type}")
+      valid_type = NotificationType.find_by_name(type)
+      raise ArgumentError, "Invalid notification type specified" if !valid_type
+      type
+    end
+
+    def create_notification(values_hash = {})
+      options = {}
+      type = notification_type(values_hash)
+      subject = notification_subject(values_hash)
+      options[:message] = values_hash[:message] if values_hash[:message].present?
+
+      _log.info("Calling Create Notification with type: #{type} subject: #{subject} options: #{options.inspect}")
+      Notification.create!(:type => type, :subject => subject, :options => options, :initiator => @workspace.ae_user)
```

Comment:
@tinaafitz 
This should get wrapped, do we need a service model for notifications. If we are not going to return the object maybe we should have a boolean (true|false) being returned from the method